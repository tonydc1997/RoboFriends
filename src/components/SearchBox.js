import React from 'react';

const SearchBox = ({ searchfield, searchChange }) => {
  return (
    <div className="pa2">
      <input
        aria-label="Search Robots"
        className="pa3 ba b--transparent mb3 bg-lightest-blue br2"
        type="search"
        placeholder="search robots"
        onChange={searchChange}
      />
    </div>
  );
};

export default SearchBox;
