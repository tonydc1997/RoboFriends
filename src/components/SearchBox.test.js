import React from 'react';
import { shallow } from 'enzyme';
import SearchBox from './SearchBox';

describe('SearchBox Component', () => {

  let searchBoxComponent;

  beforeEach(() => {
    searchBoxComponent = shallow(<SearchBox />);
  });

  it('SearchBox: returns a div and input', () => {
    expect(searchBoxComponent.html()).toContain('div');
    expect(searchBoxComponent.html()).toContain('input');
  });

  it('SearchBox: Snapshots the CSS-in-JS styles so if altered and breaks the UX, one can view the differences', () => {
    expect(searchBoxComponent).toMatchSnapshot();
  });
});
