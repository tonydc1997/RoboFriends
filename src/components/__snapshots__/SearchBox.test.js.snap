// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component SearchBox: Snapshots the CSS-in-JS styles so if altered and breaks the UX, one can view the differences 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        aria-label="Search Robots"
        className="pa3 ba b--green bg-lightest-blue"
        onChange={undefined}
        placeholder="search robots"
        type="search"
/>,
      "className": "pa2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Search Robots",
        "className": "pa3 ba b--green bg-lightest-blue",
        "onChange": undefined,
        "placeholder": "search robots",
        "type": "search",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          aria-label="Search Robots"
          className="pa3 ba b--green bg-lightest-blue"
          onChange={undefined}
          placeholder="search robots"
          type="search"
/>,
        "className": "pa2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Search Robots",
          "className": "pa3 ba b--green bg-lightest-blue",
          "onChange": undefined,
          "placeholder": "search robots",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
